uji.py

```diff
@@ -1054,6 +1059,145 @@ def upload(self):
         except Exception as e:
             logger.error(f'Failed to copy {filename}: {e}')
 
+    def clean_multi_lines_code(self):
+        import copy
+        prev_lines = self.lines
+        self.lines = copy.deepcopy(self.lines[:self.cursor_offset + 1])
```

Comment:
Yep, done. Being able to test is a good thing. I couldn't write them because the libraries we use expect a terminal and they complain with a bad ioctl or such :(